src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -0,0 +1,236 @@
+    courtOrderPoaRef: FormIF,
+    notationFormRef: FormIF
+  }
+
+  /** Prop for the item's name of the dialog. */
+  @Prop() readonly itemName: string
+
+  /** Prop for the item's filing type. */
+  @Prop() readonly filingType: string
+
+  /** Prop to display the dialog. */
+  @Prop() readonly dialog: boolean
+
+  /** Prop to provide attachment selector. */
+  @Prop() readonly attach: string
+
+  /** Prop to require court order number regardless the plan of arrangement. */
+  @Prop() readonly courtOrderNumberRequired: boolean
+
+  /** Entity number */
+  readonly entityIncNo!: string
+
+  /** The notation text. */
+  private notation: string = ''
+
+  /** Court Order Number */
+  private courtOrderNumber:string = null
+
+  /** Whether has plan of arrangement */
+  private planOfArrangement = false
+
+  /** Whether this component is currently saving. */
+  private saving = false
+
+  /** Court Order component key */
+  private courtOrderKey = 0
+
+  /** Notation form validity */
+  private notationFormValid = false
+
+  /** Notation max length */
+  private notationMaxLength = 2000
+
+  /** Flag to enable validation in this component */
+  private enableValidation = false
+
+  private get notationRules (): Array<Function> {
+    if (this.enableValidation) {
+      return [
+        (v: string) => !!v || `Enter a ${this.itemName}`,
+        (v: string) => v.length <= this.notationMaxLength || 'Maximum characters exceeded.'
+      ]
+    } else return []
+  }
+
+  private setFileNumber (courtOrderNumber: string): void {
+    this.courtOrderNumber = courtOrderNumber
+  }
+
+  private setHasPlanOfArrangement (planOfArrangement: boolean):void {
+    this.planOfArrangement = planOfArrangement
+  }
+  /** Called when prop changes (ie, dialog is shown/hidden). */
+  @Watch('dialog')
+  private async onDialogChanged (val: boolean): Promise<void> {
+    // when dialog is shown, reset notation and validation
+    if (val) {
+      this.notation = ''
+      await Vue.nextTick()
+      this.$refs.notationFormRef.resetValidation()
+
+      // This will make CourtOrderPoa to re-render as it will have a different key every time it opens the dialog.
+      // Source: https://michaelnthiessen.com/force-re-render/
+      this.courtOrderKey++
+    }
+  }
+
+  /**
+   * Emits event to close this dialog.
+   * @param needReload Whether the dashboard needs to be reloaded.
+   */
+  @Emit('close')
+  private emitClose (needReload: boolean): void {
+    this.enableValidation = false
+  }
+
+  /** WIP */
+  /** Saves the current notation. */
+  private async save (): Promise<void> {
+    // prevent double saving
+    if (this.saving) return
+
+    this.enableValidation = true
+    await this.$nextTick()
+    const isNotationFormRefValid = this.$refs.notationFormRef.validate()
+    const isCourtOrderPoaFormRefValid = this.$refs.courtOrderPoaRef.validate()
+    if (!isNotationFormRefValid || !isCourtOrderPoaFormRefValid) {
+      return
+    }
+
+    this.saving = true
```

Comment:
I think you should move this as close to the `if saving` (line 165) as possible to minimize the user's opportunity to double-click the button. Especially since you await (ie, async) on line 168, and then do work on lines 169 and 170.